File: mods/java/common/src/main/java/com/alaydriem/bedrockvoicechat/api/ConfigProvider.java
package com.alaydriem.bedrockvoicechat.api;

import com.alaydriem.bedrockvoicechat.config.ModConfig;

/**
 * Interface for platform-specific configuration loading.
 */
public interface ConfigProvider {
    /**
     * Load configuration from the platform-specific location.
     *
     * @return The loaded configuration
     */
    ModConfig load();

    /**
     * Save configuration to the platform-specific location.
     * Default implementation does nothing (not all platforms support saving).
     *
     * @param config The configuration to save
     */
    default void save(ModConfig config) {
        // Default: no-op
    }

    /**
     * Create the default configuration file if it doesn't exist.
     */
    void createDefaultIfMissing();
}

File: mods/java/common/src/main/kotlin/com/alaydriem/bedrockvoicechat/api/ConfigProvider.kt
package com.alaydriem.bedrockvoicechat.api

import com.alaydriem.bedrockvoicechat.config.ModConfig
import java.nio.file.Path

/**
 * Interface for platform-specific configuration loading.
 */
interface ConfigProvider {
    /**
     * Load configuration from the platform-specific location.
     */
    fun load(): ModConfig

    /**
     * Create the default configuration file if it doesn't exist.
     */
    fun createDefaultIfMissing()

    /**
     * Get the configuration directory path for this platform.
     * Used for embedded server mode to store certificates and database.
     * Returns null if the platform doesn't support a config directory.
     */
    fun getConfigDir(): Path?
}
